### PR TITLE
Feature/#184 sso integration test

### PR DESCRIPTION
**Pull request by charleenfei**
Addresses issue #184 .

*NOTE* some failing unit tests from interface/parameter changes. This should be addressed in a separate issue.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "test": "./node_modules/.bin/nyc -r lcov -r text ./node_modules/.bin/mocha --opts ./mocha.opts ./tests/**/*.test.ts",
     "prepublish": "rm -rf ./js/; ./node_modules/.bin/tsc -p ./tsconfig.json",
     "test:watch": "./node_modules/.bin/mocha --opts ./mocha.opts -w ./tests/**/*.test.ts",
-    "test:single": "./node_modules/.bin/nyc -r lcov -r text ./node_modules/.bin/mocha --opts ./mocha.opts"
+    "test:single": "./node_modules/.bin/nyc -r lcov -r text ./node_modules/.bin/mocha --opts ./mocha.opts",
+    "test:integration":"./node_modules/.bin/nyc -r lcov -r text ./node_modules/.bin/mocha --opts ./mocha.opts ./tests/integration/index.ts"
   },
```
Comment (nataschaberg) [inline package.json]: :+1

File: tests/integration/index.ts [hunk 2]
```diff
@@ -94,4 +100,101 @@ describe('Integration Test', () => {
...
+      issued: '1970-01-01T00:00:00.000Z',
+      proof: {
+        type: 'EcdsaKoblitzSignature2016',
+        created: '1970-01-01T00:00:00.000Z',
+        creator: 'did:jolo:5dcbd50085819b40b93efc4f13fb002119534e9374274b10edce88df8cb311af#claimId:bcf70ac9c940e',
+        nonce: '90a6764469fc4',
+        signatureValue: 'TlGz5inRY4T7GyvZi3eDq22GCEVoMlhb0mAhz7xc9y1CPe6e8PCLFK3fL4ajSD4pQhK0vFPZ3dUOdDPuopYhKQ=='
+      }
+    }]
+
```
Comment (nataschaberg) [inline tests/integration/index.ts]: Could we move the signed credentials into the respective data folder?

File: ts/interactionFlows/credentialRequest/credentialRequestPayload.ts [hunk 3]
```diff
@@ -12,16 +12,14 @@ export class CredentialRequestPayload implements IPayload {
...
-
     return credentialRequestPayload
   }
 
   public static fromJSON(json: ICredentialRequestPayloadAttrs): CredentialRequestPayload {
     const credentialRequestPayload = plainToClass(CredentialRequestPayload, json)
-    credentialRequestPayload.credentialRequest = plainToClass(CredentialRequest, json.credentialRequest)
-
+    credentialRequestPayload.credentialRequest = CredentialRequest.create(json.credentialRequest)
     return credentialRequestPayload
```
Comment (nataschaberg) [inline ts/interactionFlows/credentialRequest/credentialRequestPayload.ts]: I think we should the plainToClass logic here. Is also not working after interface adjustement?

File: tests/integration/index.ts (hunk 2 above)
Comment (charleenfei) [reply tests/integration/index.ts]: good catch

File: package.json (hunk 1 above)
Comment (katszwn) [reply package.json]: 🎉

File: tests/interactionFlows/jsonWebToken.test.ts [hunk 4]
```diff
@@ -26,6 +26,10 @@ describe('JSONWebToken', () => {
     const jsonWebToken = JSONWebToken.create(jwtCreateArgs)
 
     it('Should return a correctly assembled instance of JSONWebToken class', () => {
+      console.log(jsonWebToken.toJSON().payload.credentialRequest.credentialRequirements[0].constraints)
+      console.log('================')
+      console.log(jwtJSON.payload.credentialRequest.credentialRequirements[0].constraints)
+      console.log('================')
```
Comment (katszwn) [inline tests/interactionFlows/jsonWebToken.test.ts]: We should remove these console.logs and `only` :)

File: ts/interactionFlows/credentialRequest/credentialRequest.ts [hunk 5]
```diff
@@ -18,7 +18,10 @@ export class CredentialRequest {
   public static create(args: ICredentialRequestCreationAttrs): CredentialRequest {
     const credentialRequest = new CredentialRequest()
     credentialRequest.setCallbackURL(args.callbackURL)
-    args.credentialRequirements.forEach((req) => credentialRequest.addCredentialRequirement(req))
+    console.log(args, 'args in CR')
```
Comment (katszwn) [inline ts/interactionFlows/credentialRequest/credentialRequest.ts]: This one should go away too

File: tests/integration/index.ts [hunk 6]
```diff
@@ -1,41 +1,49 @@
...
-import { IIpfsConfig } from '../../ts/ipfs/types'
 import { IpfsStorageAgent } from '../../ts/ipfs'
 import { IEthereumResolverConfig } from '../../ts/ethereum/types'
 import { EthResolver } from '../../ts/ethereum'
 import { IdentityWallet } from '../../ts/identityWallet/identityWallet'
 import { createJolocomRegistry } from '../../ts/registries/jolocomRegistry'
 import { claimsMetadata } from '../../ts/index'
+import { JSONWebToken } from './../../ts/interactionFlows/JSONWebToken';
+import { CredentialRequest } from './../../ts/interactionFlows/credentialRequest/credentialRequest'
+import { ICredentialRequestPayloadCreationAttrs } from './../../js/interactionFlows/credentialRequest/types.d'
```
[the remark this answers is not in the document]
Comment (charleenfei) [reply tests/integration/index.ts]: fixed

**Pull request merged**